Lib/imaplib.py

```diff
@@ -1279,6 +1389,251 @@ def print_log(self):
                 n -= 1
 
 
+class Idler:
+    """Iterable context manager: start IDLE & produce untagged responses
+
+    An object of this type is returned by the IMAP4.idle() method.
+    It sends the IDLE command when activated by the 'with' statement, produces
+    IMAP untagged responses via the iterator protocol, and sends DONE upon
+    context exit.
+
+    Iteration produces (type, datum) tuples.  They slightly differ
+    from the tuples returned by IMAP4.response():  The second item in the
+    tuple is a single datum, rather than a list of them, because only one
+    untagged response is produced at a time.
+
+    Note: The name and structure of this class are internal interfaces,
+    subject to change.  Calling code can rely on its context management,
+    iteration, and public method to remain stable, but should not
+    subclass, instantiate, or otherwise directly reference the class.
+    """
+
+    def __init__(self, imap, duration=None):
+        if 'IDLE' not in imap.capabilities:
+            raise imap.error("Server does not support IMAP4 IDLE")
+        self._duration = duration
+        self._imap = imap
+        self._tag = None
+        self._sock_timeout = None
+        self._old_state = None
+
+    def __enter__(self):
+        imap = self._imap
+        assert not (imap._idle_responses or imap._idle_capture)
```

Comment:
```suggestion
        assert not imap._idle_responses
        assert not imap._idle_capture
```

To know which assertion is failing.

Reply:
Done in 055a9bd7076e.